pallets/manta-xassets/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+#![allow(unused_imports)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::marker::PhantomData;
+use cumulus_primitives_core::ParaId;
+use frame_support::{
+	pallet_prelude::*,
+	traits::{Currency, Get, Hooks, IsType, Randomness, ReservableCurrency},
+	transactional, PalletId,
+};
+use frame_system::{
+	ensure_root, ensure_signed,
+	pallet_prelude::{BlockNumberFor, OriginFor},
+};
+use sp_runtime::{
+	traits::{Saturating, Zero},
+	DispatchResult, Permill, SaturatedConversion,
+};
+use sp_std::{vec, vec::Vec};
+use xcm::v0::{
+	Error as XcmError, ExecuteXcm, Junction, MultiAsset, MultiLocation, Order, Outcome,
+	Result as XcmResult, Xcm,
+};
+use xcm_executor::{
+	traits::{Convert, FilterAssetLocation, TransactAsset, WeightBounds},
+	Assets,
+};
+
+pub use pallet::*;
+// Log filter
+const MANTA_XASSETS: &str = "manta-xassets";
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
```

Comment:
So is this a new way of writing pallets, without macros ?

Reply:
Yes.